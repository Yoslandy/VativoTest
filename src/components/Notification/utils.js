export const borders = {
    error: "border-b-semantic-error",
    warning: "border-b-semantic-warning",
    success: "border-b-semantic-success",
}

export const colors = {
    error: "text-semantic-error",
    warning: "text-semantic-warning",
    success: "text-semantic-success",
}
