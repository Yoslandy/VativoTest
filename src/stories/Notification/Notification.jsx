import React from "react"

import Notification from "../../components/Notification"

// eslint-disable-next-line react/prop-types
export default function CustomNotification({ appearance, children, title }) {
    return (
        <Notification appearance={appearance} title={title}>
            {children}
        </Notification>
    )
}
