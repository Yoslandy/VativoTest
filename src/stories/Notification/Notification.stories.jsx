import React from "react"

import "@fortawesome/fontawesome-free/css/all.min.css"
import "../../assets/styles/build.css"

import Notification from "./Notification"

export default {
    title: "Example/Notification",
    component: Notification,
    argTypes: {
        appearance: "success",
        title: "Title",
        children: "Notification Title",
    },
}

function Template(args) {
    return <Notification {...args} />
}

export const Success = Template.bind({})
Success.args = {
    appearance: "success",
    title: "Success",
    children: "Notification Title",
}

export const Warning = Template.bind({})
Warning.args = {
    appearance: "warning",
    title: "Warning",
    children: "Notification Title",
}

export const Error = Template.bind({})
Error.args = {
    appearance: "error",
    title: "Error",
    children: "Notification Title",
}
