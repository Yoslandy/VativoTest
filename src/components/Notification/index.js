/* eslint-disable react/prop-types */
import React, { useCallback } from "react"
import PropTypes from "prop-types"
import { borders, colors } from "./utils"
import Icon from "../Icon"
import SuccessIcon from "../../assets/img/notification-icons/succes.png"

export default function Notification(props) {
    const { appearance, onDismiss, children, title } = props

    const getColorIcon = useCallback(() => colors[appearance], [appearance])

    return (
        <div
            className={`border-b-8 bg-white rounded w-72 p-2 h-full shadow ${borders[appearance]}`}
        >
            <div className="flex w-full">
                <div>
                    {appearance === "success" ? (
                        <img src={SuccessIcon} alt="success" />
                    ) : (
                        <Icon type={appearance} className={getColorIcon()} />
                    )}
                </div>
                <div className="flex flex-col ml-4">
                    <p className="font-medium">{title}</p>
                    <span className="font-normal">{children}</span>
                </div>
                <span onClick={onDismiss} className="absolute right-[27px]">
                    <i className="fas fa-times ml-8 cursor-pointer" />
                </span>
            </div>
        </div>
    )
}

Notification.propTypes = {
    title: PropTypes.string,
    appearance: PropTypes.oneOf(["error", "success", "warning"]),
}

Notification.defaultProps = {
    title: "Title",
    appearance: "success",
}
