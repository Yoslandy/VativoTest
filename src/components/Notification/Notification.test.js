import React from "react"
import ReactDOM from "react-dom"
import { MemoryRouter } from "react-router-dom"
import renderer from "react-test-renderer"
import { ToastProvider } from "react-toast-notifications"
import Notification from "./index"

describe("Notification", () => {
    beforeAll(() => {
        ReactDOM.createPortal = jest.fn((element, node) => {
            return element
        })
    })

    test("Renders Notification correctly", () => {
        const component = renderer.create(
            <MemoryRouter>
                <ToastProvider
                    portalTargetSelector="body"
                    components={{ Toast: Notification }}
                ></ToastProvider>
            </MemoryRouter>
        )

        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })

    afterEach(() => {
        ReactDOM.createPortal.mockClear()
    })
})
