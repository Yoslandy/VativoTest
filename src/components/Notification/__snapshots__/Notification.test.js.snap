// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification Renders Notification correctly 1`] = `null`;
